packages/react-ui/src/features/flows/components/flow-actions-menu.tsx

```diff
@@ -0,0 +1,119 @@
+import { FlowOperationType, PopulatedFlow } from '@activepieces/shared';
+import { useMutation } from '@tanstack/react-query';
+import {
+  Copy,
+  Download,
+  EllipsisVertical,
+  Pencil,
+  Share2,
+  Trash2,
+} from 'lucide-react';
+import React from 'react';
+
+import { flowsApi } from '../lib/flows-api';
+import { flowsUtils } from '../lib/flows-utils';
+
+import { ConfirmationDeleteDialog } from '@/components/delete-dialog';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from '@/components/ui/dropdown-menu';
+import { INTERNAL_ERROR_TOAST, toast } from '@/components/ui/use-toast';
+import { authenticationSession } from '@/lib/authentication-session';
+
+interface FlowActionMenuProps {
+  flow: PopulatedFlow;
+  onRename: (flow: PopulatedFlow) => void;
+  onDuplicate: () => void;
+  onShare: (flow: PopulatedFlow) => void;
+  onDelete: () => void;
+}
+
+const FlowActionMenu: React.FC<FlowActionMenuProps> = ({
+  flow,
+  onRename,
+  onDuplicate,
+  onShare,
+  onDelete,
+}) => {
+  const { mutate: duplicateFlow } = useMutation({
+    mutationFn: async () => {
+      const createdFlow = await flowsApi.create({
+        displayName: flow.version.displayName,
+        projectId: authenticationSession.getProjectId(),
+      });
+      const updatedFlow = await flowsApi.update(createdFlow.id, {
+        type: FlowOperationType.IMPORT_FLOW,
+        request: {
+          displayName: flow.version.displayName,
+          trigger: flow.version.trigger,
+        },
+      });
+      return updatedFlow;
+    },
+    onSuccess: (data) => {
+      window.open(`/flows/${data.id}`, '_blank', 'rel=noopener noreferrer');
+      onDuplicate();
+    },
+    onError: () => toast(INTERNAL_ERROR_TOAST),
+  });
+
+  const exportFlow = async () => {
```

Comment:
I would move api request into downloadFlow, and then use mutate to handle onSuccess and response and LoadingState